packages/glimmer-runtime/lib/compiled/opcodes/content.ts

```diff
@@ -27,12 +39,20 @@ export class AppendOpcode extends Opcode {
   evaluate(vm: VM) {
     let reference = vm.frame.getOperand();
 
-    let mapped = map(reference, normalizeTextValue);
+    let mapped = map(reference, normalizeTextOrTrustedValue);
     let cache = new ReferenceCache(mapped);
-    let node = vm.stack().appendText(cache.peek());
-
-    if (!isConst(reference)) {
-      vm.updateWith(new UpdateAppendOpcode(cache, node));
+    let value = cache.peek();
+
+    if (isSafeString(value)) {
+      let bounds = vm.stack().insertHTMLBefore(null, value);
```

Comment:
we should handle `isConst` for this branch also (which argues for better factoring of the code) – it could happen if you have a helper that produces a SafeString and we noticed that the inputs to the helper is static